### PR TITLE
Setting IHttpContextAccessor in CreateHttpContext #405

### PR DESCRIPTION
**Pull request by JunTaoLuo**
#405

### CODE REVIEW

File: src/Microsoft.AspNet.Hosting/Builder/HttpContextFactory.cs [hunk 1]
```diff
@@ -9,9 +9,18 @@ namespace Microsoft.AspNet.Hosting.Builder
...
+        public HttpContextFactory(IHttpContextAccessor httpContextAccessor)
+        {
+            _httpContextAccessor = httpContextAccessor;
+        }
+
         public HttpContext CreateHttpContext(IFeatureCollection featureCollection)
         {
-            return new DefaultHttpContext(featureCollection);
+            var httpContext = new DefaultHttpContext(featureCollection);
+            _httpContextAccessor.HttpContext = httpContext;
```
Comment (benaadams) [inline src/Microsoft.AspNet.Hosting/Builder/HttpContextFactory.cs]: Should this be
```csharp
if (_httpContextAccessor != null) 
{
    _httpContextAccessor.HttpContext = httpContext;
}
```

Comment (JunTaoLuo) [inline src/Microsoft.AspNet.Hosting/Builder/HttpContextFactory.cs]: Yes that would be a good idea.

File: src/Microsoft.AspNet.Hosting/Builder/HttpContextFactory.cs [hunk 2]
```diff
@@ -9,9 +9,23 @@ namespace Microsoft.AspNet.Hosting.Builder
...
+        {
+            _httpContextAccessor = httpContextAccessor;
+        }
+
         public HttpContext CreateHttpContext(IFeatureCollection featureCollection)
         {
-            return new DefaultHttpContext(featureCollection);
+            var httpContext = new DefaultHttpContext(featureCollection);
+
+            if (_httpContextAccessor != null)
```
Comment (davidfowl) [inline src/Microsoft.AspNet.Hosting/Builder/HttpContextFactory.cs]: Actually we don't support null constructor injection so we may need to adjust that

**Comment by Tratcher** [comment]
:shipit:

**Comment by davidfowl** [comment]
:shipit:

**Comment by JunTaoLuo** [comment]
I merged this to dev even though tests failed since this is a small change and travis won't get the updated packages fixes until late today or early tomorrow when coherence passes again. The tests previously passed on AppVeyor.

**Pull request merged**